### PR TITLE
Added cache and coverall

### PR DESCRIPTION
**Pull request by aneeshkp**

### CODE REVIEW

File: internal/resourcecache/resourcecache.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	KindTypeRole                = "Role"
+	KindTypeSecret              = "Secret"
+	KindTypeVolume              = "Volume"
+	KindTypeDaemonSet           = "DaemonSet"
+	KindTypePod                 = "Pod"
+	KindTypeContainer           = "Container"
+)
+
+/*
+ResourceCache -->cache["ROLE"]->Resourc{package:roles,filename:deployment.go,functions:[]Functions}
```
Comment (sjakati98) [inline internal/resourcecache/resourcecache.go]: Comment filename should match package name

File: internal/resourcecache/resourcecache.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+func (r *ResourceCache) GetFunctionsByKind(kind KindType) []ResourceFunction {
+	return r.cache[kind].Functions
+}
+
+//GetKindType ...
+func (r *ResourceCache) GetKindType(kind KindType) (*Resource, error) {
+	if value, ok := r.cache[kind]; ok {
+		return value, nil
+	}
+	return nil, errors.New("Key not found ")
```
Comment (abays) [inline internal/resourcecache/resourcecache.go]: Remove extra space and add a dot.

**Review by abays — APPROVED** [review]

**Pull request merged**